Apple/NetworkExtension/DataTypes.swift

```diff
@@ -0,0 +1,39 @@
+import Foundation
+
+enum BurrowError: Error {
+    case addrDoesntExist
+    case resultIsError
+    case cantParseResult
+}
+
+protocol Request: Codable {
+    var id: UInt { get set }
+    var command: String { get set }
+}
+
+struct BurrowRequest: Request {
+    var id: UInt
+    var command: String
+}
+
+struct Response<T>: Decodable where T: Decodable {
+    var id: UInt
+    var result: T
+}
+
+// swiftlint:disable identifier_name
+struct BurrowResult<T>: Codable where T: Codable {
+    var Ok: T?
+    var Err: String?
+}
```

Comment:
Ha! We should change this to not use the Rust labels at some point. Not blocking for now, though